### DISCUSSION TITLE
Basic: Configure Weblate to see sentences in context?

### DISCUSSION THREAD

**Discussion by oslogrolls** [WeblateOrg/weblate#11441] [Q&A] [answered]
I'm a beginner at translation management and had a look at Weblate's sample projects. Only getting offered short phrases and single sentences for translation was very irritating to me, but I learned that this approach is typical for professional level translation software.

 I don't doubt the value of this approach for efficiency – yet, the missing context looks very foreign to me. Is there an alternative view-mode available, that shows sentences in their context?

**Comment by nijel** [comment]
You can see nearby strings in the bottom of the translating interface.

**Comment by emmapeel2** [comment]
To provide context, in the Tor Project websites we also use the 'Source string location' links to see the string in context (in the English version on production). See for example: https://hosted.weblate.org/translate/tor/tor-browser/tor-browser-user-manual/af/?checksum=88fa832bd69bbc6a (on the sidebar)

This is unfortunately not possible in all formats.

This is the line I have added on Project settings - Version control - Repository browser:

https://gitlab.torproject.org/tpo/web/support/-/blob/main/content/{{filename|parentdir}}.lr

to get the link to edit the source page.

**Comment by oslogrolls** [comment]
Thank you, both! From my question, you can see that I am a newbie with professional translation software. Actually, it seems our background is not typical for Weblate – most of the Weblate projects I've looked at are web applications or software interfaces and their documentation.

Our use case would be E-Commerce – we would mainly translate **product-descriptions**. It is crucial to us that translations are not only factually and grammatically correct, but also convey our tone of voice. We therefore do not want to translate _sentence by sentence_, but would like to see the entire original text or paragraph in context – even if this might be regarded _uneconomical_ among professional translators.
 
Perhaps this is possible with appropriate configuration? We had to try out if linking to the original resource would work for us.

**Reply by nijel** [reply, marked as answer]
Weblate offers whatever translation units that are stored in the translation files (or uploaded to Weblate), so it's up to you to define units you want to translate. There is always trade-offs when translating long texts: 

* short units lack context and can end up being messy when concatenated
* long units need to be fully retranslated after a minor source change

**Reply by oslogrolls** [reply]
Thanks, Michal, that was helpful.
